sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -206,6 +206,10 @@ struct _pi_object {
   // Level Zero doesn't do the reference counting, so we have to do.
   // Must be atomic to prevent data race when incrementing/decrementing.
   std::atomic<pi_uint32> RefCount;
+
+  // Protects accesses to all the non-const member variables.  Exclusive access
+  // is required to modify any of these members.
```

Comment:
1) Please add examples here how to properly lock things (including multiple objects, and read/write), so that people use the same syntax more or less.

2) I think we should add a mode where single-threaded app doesn't pay the overhead of working with many mutexes. Please see how this can get organized. Maybe wrap the "std::shared_mutex" into a class that would redefine mutex operations to no-op under some env var.

Reply:
Added examples and single-threaded mode.